text/0000-track-unique-history-location-state.md

```diff
@@ -0,0 +1,93 @@
+- Start Date: 2016/12/05
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Track unique history location states
+
+# Motivation
+
+The path alone does not provide enough information. For example, if you
+visit page A, scroll down, then click on a link to page B, then click on
+a link back to page A. Your actual browser history stack is [A, B, A].
+Each of those nodes in the history should have their own unique scroll
+position. In order to record this position we need an ID that is unique
+for each node in the history.
+
+# Detailed design
+
+Code: [PR#14011](https://github.com/emberjs/ember.js/pull/14011)
+
+We simply add a `stateCounter` so we can track uniqueness on two
```

Comment:
`stateCounter` i assume returns a number, do we wan to also guarantee that `stateCounter` is always incremented, such as `counterFor(state1) < counterFor(state2) < counterFor(stateN)`?

If it is guaranteed to be incrementing, is it incrementing for the life of the page, for the life of the app or for the life of the location.  (this is more something i thought of, when thinking about how acceptances tests might interact with one another.

Alternatively, we can simply say it is a session unique id for a given application instance, and in practice it could be represented by a number, or a guid or....